lib/Auth/Process/CheckIsCesnetEligible.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace SimpleSAML\Module\cesnet\Auth\Process;
+
+use DateTime;
+use SimpleSAML\Auth\ProcessingFilter;
+use SimpleSAML\Auth\State;
+use SimpleSAML\Configuration;
+use SimpleSAML\Locale\Translate;
+use SimpleSAML\Logger;
+use SimpleSAML\Module;
+use SimpleSAML\Module\perun\Exception;
+use SimpleSAML\Session;
+use SimpleSAML\Utils\HTTP;
+
+/**
+ * Class CheckIsCesnetEligible
+ * @package SimpleSAML\Module\cesnet\Auth\Process
+ *
+ * @author Pavel Vyskocil <pavel.vyskocil@cesnet.cz>
+ */
+class CheckIsCesnetEligible extends ProcessingFilter
+{
+    const PAGE_HEADER = '{cesnet:einfra:check_cesnet_eligible-header}';
+    const PAGE_TEXT_OLD_VALUE = '{cesnet:einfra:check_cesnet_eligible-old_value_text}';
+    const PAGE_TEXT_NONE_VALUE = '{cesnet:einfra:check_cesnet_eligible-none_value_text}';
+    const IS_CESNET_ELIGIBLE = 'isCesnetEligibleLastSeen';
+
+    public function __construct($config, $reserved)
+    {
+        parent::__construct($config, $reserved);
+    }
+
+    public function process(&$request)
+    {
+        assert('is_array($request)');
+
+//        Logger::debug(print_r($request, true));
+
+        // Get SP required attributes
+        $requiredAttributes = [];
+        if (isset($request['Destination']['attributes'])) {
+            $requiredAttributes = $request['Destination']['attributes'];
+        }
+
+        if (!in_array(self::IS_CESNET_ELIGIBLE, $requiredAttributes)) {
+            Logger::debug('cesnet:CheckIsCesnetEligible - SP didn\'t request attribute \'isCesnetEligible\'. 
+            Skipping to next filter. ');
+            return;
+        }
+
+        $isCesnetEligibleLastSeenString = null;
+        if (isset($request['Attributes'][self::IS_CESNET_ELIGIBLE])) {
+            $isCesnetEligibleLastSeenString = $request['Attributes'][self::IS_CESNET_ELIGIBLE][0];
+        } else {
+            Logger::debug('cesnet:CheckIsCesnetEligible - SP requested attribute \'isCesnetEligible\', but user 
```

Comment:
change text: 

'cesnet:CheckIsCesnetEligible - SP requested attribute \'isCesnetEligible\', but user didn\'t have any value in attrubute. => Redirected to unauthorized page.');